index.js

```diff
@@ -25,6 +25,16 @@ module.exports = {
     }
   },
 
+  files() {
+    let files = this._super.files.apply(this, arguments);
+
+    if (this.project.pkg.name) {
```

Comment:
in order to make this easier to modify in the future lets move it to a method that we can easily iterate on. Possible suggestion:

```js
if (this._shouldIncludeYarnLockInFiles()) {
}
```